index.js

```diff
@@ -0,0 +1,103 @@
+//set requires
+const express = require("express");
+const cors = require("cors");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+
+const db = require("./helpers/dbHelpers");
+
+//set port #
+const PORT = 9000;
+
+const server = express();
+server.use(express.json(), cors());
+
+//import secret for JWT
+const secret = "secretsarenofununlessyoutelleveryone";
+
+//restricts access based on login
+function protect(req, res, next) {
+   const token = req.headers.authorization;
+   if(token){
+      jwt.verify(token, secret, (err, decodedToken) => {
+         if(err) {
+            res.status(400).send("You shall not pass! Invalid Token")
+         } else {
+//not sure what this does need to ask about this tuesday**
+            req.username = decodedToken.username
```

Comment:
If there is no error, it will match with the username that the person is trying to log in with.